File: BestTimeToBuyandSell/BTTBSI.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profits = [0 for i in range(0,len(prices))]
        for i in range(0,len(prices)-1):
            profits[i] = max(prices[i:]) - prices[i] 
        return 0 if len(prices)<=1 else max(profits)

class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minp,maxp = float('inf'),0
        for price in prices:
            minp = min(price,minp)
            profit = price-minp
            maxp = max(profit,maxp)
        return maxp
    

File: BestTimeToBuyandSell/BTTBSI.java
class Solution {
    public int maxProfit(int[] prices) {
        if(prices.length == 0) return 0;
        int maxp = 0;
        int minp = prices[0];
        for(int price : prices){
            minp = Math.min(price,minp);
            int curr = price - minp;
            maxp = Math.max(curr,maxp);
        }
        return maxp;
    }
}